clients/drcachesim/analyzer.cpp

```diff
@@ -274,12 +291,32 @@ analyzer_tmpl_t<RecordType, ReaderType>::get_error_string()
     return error_string_;
 }
 
+template <typename RecordType, typename ReaderType>
+bool
+analyzer_tmpl_t<RecordType, ReaderType>::advance_interval_id(
+    analyzer_worker_data_t *worker, const RecordType &record,
+    uint64_t &prev_interval_index)
+{
+    if (interval_microseconds_ == 0 || !record_is_timestamp(record)) {
+        return false;
+    }
+    uint64_t next_interval_index =
+        (worker->stream->get_last_timestamp() - worker->stream->get_first_timestamp()) /
```

Comment:
This is the stream from the scheduler potentially containing multiple shards mapped to one worker: we need to have per-shard timestamps.  (Even if the scheduler guaranteed to run one to completion before the next, which it does not, the get_first_timestamp() will be incorrect as it will point to the first shard and not the current shard.)

Reply:
Fixed the get_first_timestamp impl in scheduler using SCHEDULER_USE_INPUT_ORDINALS.